tls/extensions/s2n_extension_type.c

```diff
@@ -157,6 +158,11 @@ int s2n_extension_never_send(struct s2n_connection *conn)
     return false;
 }
 
+int s2n_extension_send_if_tls13_enabled(struct s2n_connection *conn)
```

Comment:
i'll be very careful about using this as a generic method.

for supported version extension, i would also check check if the security configuration uses tls13.

elsewhere, i would check if conn->actual_protocol_version == tls13